src/sections/collection.js

```diff
@@ -0,0 +1,259 @@
+import "./styles/collection.scss";
+import { onDocumentReady } from "../utils/dom";
+import { addToCart } from "../utils/addToCart";
+
+const state = {
+  elements: {},
+};
+
+const cacheState = () => {
+  state.elements = {
+    sortBy: document.getElementById("SortBy"),
+    collectionFilterCheckbox: document.querySelectorAll(
+      ".collectionFilter input[type='checkbox']",
+    ),
+    collectionFilterPrice: document.querySelectorAll(
+      ".filter-group-display__price-range input[type='number']",
+    ),
+    paginationItems: document.querySelectorAll("[data-pagination-item]"),
+    paginationLinks: document.querySelectorAll("[data-pagination-link]"),
+    paginateNext: document.querySelector("[data-next]"),
+    collectionGrid: document.getElementById("CollectionGrid"),
+    collectionButton: document.querySelector(".collection__load--button"),
+    filterButton: document.querySelector(".collection__filter--button"),
+    filterWrapper: document.querySelector(".collection__filters"),
+    collectionOverlay: document.querySelector(".collection__overlay"),
+    body: document.querySelector("body"),
+    filterCloseButton: document.querySelector(
+      ".collection__filters .icons__wrapper--span",
+    ),
+    paginateBtn: document.querySelectorAll(".collection__paginate--btn"),
+    iconGrid: document.querySelector(".icon__grid"),
+    iconList: document.querySelector(".icon__list"),
+    collectionProduct: document.querySelectorAll(".collection__product"),
+    url: new URL(window.location.href),
+    sortResult: "created-ascending",
+    defaultPage: 2,
+    filterResult: "",
+    filterPriceResult: "",
```

Comment:
state.elements should contains only DOM elements and such data can be stored directly in state